### ISSUE TITLE
Loading a model in a separate function 

### ISSUE THREAD

**Issue by sfilatye** [fchollet/keras#3193]
I am calling a function to load a model 

def read_model(index, cross='',img_rows1=244, img_cols1=244):
    model=vgg_std16_model(img_rows1, img_cols1, color_type_global)
    model.load_weights('cache/model_weights2_vgg_16_3x4.h5')
    sgd = SGD(lr=1e-3, decay=1e-6, momentum=0.9, nesterov=True)
    model.compile(optimizer=sgd, loss='categorical_crossentropy')
    return model
 
The function is being used somewhere else as:

model = read_model(index, modelStr)
test_prediction = model.predict(test_data, batch_size=batch_size, verbose=1)

I get an error: "Exception: Error when checking : expected zeropadding2d_input_1 to have shape (None, 3, 244, 244) but got array with shape (1000, 3, 224, 224)"

However, the error disappears if I call the function in this way:
img_rows, img_cols = 224, 224
model = read_model_new2(index, modelStr, img_rows, img_cols )
test_prediction = model.predict(test_data, batch_size=batch_size, verbose=1)

What's going on?

**Comment by mbollmann** [comment]
224 and 244 are not the same numbers.

**Comment by sfilatye** [comment]
Thanks!
